docs/_tutorials/read-only-json-api.md

```diff
@@ -0,0 +1,416 @@
+by [Régis Philibert](https://twitter.com/regisphilibert). It's about doing the exact same thing in
+[Hugo](https://gohugo.io/).
+
+I hope that this solution in Jekyll is elegant, too. You will need to define a number of [layouts]({% link _docs/layouts.md %})
+and write a [plugin]({% link _docs/plugins.md %}) to convert your content to JSON. **Don't be afraid if you're not
+familiar with Ruby** --- the language that Jekyll is written in. It's similar to Python, and you should get a hang of it
+quickly.
+
+By the end of the tutorial, you'll expose the following API:
+* `/` --- an index of all posts
+* `/{url}` --- details of a post
+* `/categories` --- an index of categories
+* `/categories/{category}` --- an index of posts under a category
+
+Let's dive in!
+
+{: .note .info}
+This solution needs a custom environment to support its Ruby code. This means that you can't host the site on GitHub
+Pages as-is - you'd need to use [Actions]({% link _docs/continuous-integration/github-actions.md %}) to deploy your
+site. First, see if a simpler [solution](https://gist.github.com/MichaelCurrin/f8d908596276bdbb2044f04c352cb7c7)
+created by [Michael Currin](https://github.com/MichaelCurrin) fits your needs.
+
+## Setup
+
+First, you need to go through an [installation]({% link _docs/installation.md %}) process. Having done that, you can
+initialize a new Jekyll project by invoking the command `bundle exec jekyll new`. 
+
+### Directories Structure
+
+You need to create the following structure:
+
+```
+.
+├── site
+│   ├── config.yml
+│   ├── _includes
+│   │   └── post.html
+│   ├── _layouts
+│   │   ├── categories_index.html
+│   │   ├── items_index.html
+│   │   └── post.html
+│   ├── _posts
+│   │   ├── 2020–12–20-the-first-post.md
+│   │   └── 2020–12–20-the-second-post.md
+```
+
+Unless you self-host your site, **you can rely on a generic 404 response from your server**. You no longer need `index`
+ --- you will generate a custom one later. Generating pages like `about` is out of scope for this tutorial, but in case
+you need them, they behave the same way that posts do.
+
+### Dummy Posts
+
+Under `_posts`, create two new documents. Remember that Jekyll requires you to **prepend post filenames with a date**:
+
+`2020–12–20-the-first-post.md`
+
+```md
+---
+title:  "The First Post"
+category: update
+custom-property: "My custom property"
+---
+1
+```
+
+Make sure to include a `custom-property` so that you can fetch it later on. 
+
+`2020–12–20-the-second-post.md`
+
+```md
+---
+title:  "The Second Post"
+categories: ["update", "tutorial"]
+---
+2
+```
+
+This one belongs to two categories --- `update` and `tutorial` --- and doesn't include the custom property. Because the
+categories overlap, it will be possible to test getting a list of posts from a category.
+
+### Layouts
+
+The last thing is to define the layouts. First, declare them in `config.yml`:
```

Comment:
Too early to talk about doing *the last thing*. We haven't created the Ruby plugin yet.
Besides, using front matter defaults and new types will only be clear after the Ruby plugin has been implemented.

So it's best to move this section to the bottom.

Reply:
Done. I also made some changes to this section, so that it reflects the fact that it's the last thing I describe.